File: data structures/stack/python/stack_test.py
import unittest
import stack

class StackTest(unittest.TestCase):

    def setUp(self):
        self.stack = Stack()

    def test_can_push(self):
        self.stack.push(1)
        self.assertFalse(self.stack.isEmpty())
        
    def test_is_empty(self):
        self.assertTrue(self.stack.isEmpty())
        self.stack.push(23)
        self.assertFalse(self.stack.isEmpty())
        
    def test_error_when_pop_empty(self):
        with self.assertRaises(AssertionError):
            self.stack.pop()
            
    def test_multiple_push_and_pop(self):
        elements = ['tea', 42, 'roof']
        
        for element in elements:
            self.stack.push(element)
        
        elements.reverse()
        for element in elements:
            self.assertEqual(element, self.stack.pop())
        
        
if __name__ == "__main__": 
    unittest.main()

File: data structures/stack/ruby/stack_test.rb
require 'test/unit'
require_relative 'stack'

class StackTest < Test::Unit::TestCase

  # Called before every test method runs. Can be used
  # to set up fixture information.
  def setup
    @sut = Stack.new
  end

  def test_can_push_elements
    @sut.push 1
    assert_false @sut.empty?
  end
  
  def test_can_pop
    element = 'foo'
    @sut.push element
    assert_equal element, @sut.pop
  end

  def test_cannot_pop_on_empty
    assert_raises { @sut.pop }
  end

  def test_can_push_multiple
    values = ['foo', nil, :bar, 42]
    values.each { |element| @sut.push element }
    values.reverse.each do |element|
      assert_equal element, @sut.pop
    end
  end
end